### PR TITLE
BloomFilterRule should only recommend for supported column type

### PR DESCRIPTION
**Pull request by yuanbenson**
Instructions:
1. The PR has to be tagged with at least one of the following labels (*):
   1. `feature`
   2. `bugfix`
   3. `performance`
   4. `ui`
   5. `backward-incompat`
   6. `release-notes` (**)
2. Remove these instructions before publishing the PR.
 
(*) Other labels to consider:
- `testing`
- `dependencies`
- `docker`
- `kubernetes`
- `observability`
- `security`
- `code-style`
- `extension-point`
- `refactor`
- `cleanup`

(**) Use `release-notes` label for scenarios like:
- New configuration options
- Deprecation of configurations
- Signature changes to public methods/interfaces
- New plugins added or old plugins removed

### CODE REVIEW

**Comment by yuanbenson** [comment]
@GSharayu please take a look. I'm not sure why I can't add you as reviewers directly.

File: pinot-controller/src/main/java/org/apache/pinot/controller/recommender/rules/impl/BloomFilterRule.java [hunk 1]
```diff
@@ -41,6 +45,8 @@
 public class BloomFilterRule extends AbstractRule {
   private static final Logger LOGGER = LoggerFactory.getLogger(BloomFilterRule.class);
   private final BloomFilterRuleParams _params;
+  private static final Set<DataType> COMPATIBLE_DATA_TYPES = new HashSet<>(
+      Arrays.asList(DataType.INT, DataType.LONG, DataType.FLOAT, DataType.DOUBLE, DataType.STRING, DataType.BYTES));
```
Comment (yuanbenson) [inline pinot-controller/src/main/java/org/apache/pinot/controller/recommender/rules/impl/BloomFilterRule.java:49]: Assumed these are the compatible types after looking at BloomFilterHandler.java

Comment (GSharayu) [reply pinot-controller/src/main/java/org/apache/pinot/controller/recommender/rules/impl/BloomFilterRule.java:49]: (nit) you want to add this as comment as well?

**Comment by GSharayu** [comment]
Feel free to open the PR. LGTM

**Pull request merged**